src/sentry/event_manager.py

```diff
@@ -1884,11 +1884,38 @@ def _create_group(project: Project, event: Event, **group_creation_kwargs: Any)
 
     group_creation_kwargs["data"] = group_data
 
-    return Group.objects.create(
-        project=project,
-        short_id=short_id,
-        **group_creation_kwargs,
-    )
+    try:
+        with transaction.atomic(router.db_for_write(Group)):
+            # This is the 99.999% path. The rest of the function is all to handle a very rare and
+            # very confounding bug which keeps projects from creating new groups.
+            group = Group.objects.create(
+                project=project,
+                short_id=short_id,
+                **group_creation_kwargs,
+            )
+
+    # Attempt to handle The Mysterious Case of the Stuck Project Counter
+    except IntegrityError as err:
+        if not _is_stuck_counter_error(err, project, short_id):
+            raise
+
+        new_short_id = _handle_stuck_project_counter(project, short_id)
```

Comment:
I'm a bit worried about race conditions here. Although I guess if two processes do the fix the worst case is probably that we just increment too much.

The other option is to lock around this. But then if we fail to acquire the lock, then that group creation fails, unless we retry a few times...

Reply:
Yeah - hard to anticipate every eventuality. It's entirely kicking the can down the road, but given how comparatively rarely we're going to use this codepath, and then how _more_ rarely we'll be in this code path with two events which create different new groups simultaneously... can we just say, we'll deal with that error when we get there? I'm happy to add a comment about the possibility of a race condition, so that if it does come up, it'll be a breadcrumb for whoever investigates it, but maybe we just don't worry about it for now? In the interest of getting this fix out the door?

WDYT?